tests/e2e/utils/track-event.js

```diff
@@ -10,14 +10,27 @@
  *
  * @param {Page} page
  * @param {string} eventName Event name to match.
+ * @param {string|null} URLpath The URL path where the event should be triggered.
```

Comment:
I don't think we need to over engineer this function but maybe add a better description that it matches the starting part of the path.

For example if we go to checkout: `https://domain.test/checkout/`
When we complete checkout we are redirected to: `https://domain.test/checkout/order-received/<orderId>/`

Both of those would match if URLpath is set to `checkout`